### PR TITLE
fix(jira-syntax): correct {code}/{panel} closing detection + remove inline-code suppression

### PR DESCRIPTION
**Pull request by powli**
## Summary

Two cherry-picks from the upstream fork [`tgreat1/jira-skill`](https://github.com/tgreat1/jira-skill) fixing two false-positive bugs in `validate-jira-syntax.sh`, plus one consistency refactor and a mode-bit restoration.

## Cherry-picked fixes

- **`{code}`/`{panel}` closing-tag detection** — cherry-picked from [`8255a1b`](https://github.com/tgreat1/jira-skill/commit/8255a1bd7859f4a0b54950dc7895f5ced0f9a723).
  The validator was looking for `{/code}` and `{/panel}` (XML-style closers) which never appear in valid Jira wiki markup — causing **every** file with a code block to fail validation. Replaces the open-vs-close count diff with an even-odd occurrence check (Jira uses the same tag for open and close).
- **Inline-code warning suppression** — cherry-picked from [`c0e3e3e`](https://github.com/tgreat1/jira-skill/commit/c0e3e3e6acae1fd851b631657c517a1307c507fe).
  Previously: backtick-inline-code warnings were silently suppressed when *any* `{{...}}` substitution existed in the file. Now the warning fires on backticks regardless.

## Additional commits on this branch

- **Restore executable bit** — cherry-picking `8255a1b` dropped the file mode from `100755` to `100644`. Restored.
- **Use occurrence count, not line count** — tightens the parity check from `grep -c` (line count) to `grep -o ... | wc -l` (occurrence count), matching the existing `{color}` check on the same script. Catches edge cases like `{code} inline {code} {code}` (3 tags on one line) that line-count would miss.

## Test plan

- [x] `pytest tests/` — 507 pass
- [x] `ruff check .` — clean
- [x] `ruff format --check .` — clean (39 files)
- [x] `bandit -r skills/jira-communication/scripts/ scripts/ -c pyproject.toml --severity-level medium` — 0 findings
- [x] `bash scripts/verify-harness.sh --status` — Level 3 (Enforced) COMPLETE
- [x] Manual: `validate-jira-syntax.sh` on a balanced `{code}...{code}` file passes (previously errored); on an unbalanced single `{code}` opener still errors with "odd number (1) of occurrences".

Cherry-pick attribution preserved via `git cherry-pick -x` trailers.

### CODE REVIEW

File: skills/jira-syntax/scripts/validate-jira-syntax.sh [hunk 1]
```diff
@@ -75,7 +75,7 @@ validate_file() {
     fi
 
     # Check for Markdown-style inline code (` instead of {{)
-    if echo "$content" | grep -qE "\`[^\`]+\`" && ! echo "$content" | grep -qE "{{[^}]+}}"; then
+    if echo "$content" | grep -qE "\`[^\`]+\`"; then
         warning "Found Markdown inline code (\`code\`). Consider Jira format: {{code}}"
```
[the remark this answers is not in the document]
Comment (powli) [reply skills/jira-syntax/scripts/validate-jira-syntax.sh]: Fixed in 9bcfa49: replaced `echo` pipeline with bash here-string (`<<< "$content"`) on the inline-code check.

File: skills/jira-syntax/scripts/validate-jira-syntax.sh [hunk 2]
```diff
@@ -75,7 +75,7 @@ validate_file() {
     fi
 
     # Check for Markdown-style inline code (` instead of {{)
-    if echo "$content" | grep -qE "\`[^\`]+\`" && ! echo "$content" | grep -qE "{{[^}]+}}"; then
+    if echo "$content" | grep -qE "\`[^\`]+\`"; then
```
[the remark this answers is not in the document]
Comment (powli) [reply skills/jira-syntax/scripts/validate-jira-syntax.sh]: Fixed in 9bcfa49: replaced `echo "$content" | grep ...` with `grep ... <<< "$content"` here-string on the inline-code check (line 78).

File: skills/jira-syntax/scripts/validate-jira-syntax.sh [hunk 3]
```diff
@@ -104,17 +104,21 @@ validate_file() {
...
-    local code_open=$(echo "$content" | grep -c "{code")
-    local code_close=$(echo "$content" | grep -c "{/code}")
-    if [ "$code_open" -ne "$code_close" ]; then
-        error "Mismatched {code} tags: $code_open opening, $code_close closing"
+    # Jira wiki markup uses {code} as both the opening and closing tag, so a
+    # correctly paired block always produces an even occurrence count.
+    # Use `grep -o ... | wc -l` to count each occurrence (not just matching
+    # lines), matching the {color} check below for consistency and to catch
+    # multiple tags on the same line.
+    local code_count=$(echo "$content" | grep -o "{code" | wc -l)
```
[the remark this answers is not in the document]
Comment (powli) [reply skills/jira-syntax/scripts/validate-jira-syntax.sh]: Fixed in 9bcfa49: switched the {code} count to a here-string (`<<< "$content"`) and tightened the regex to `\{code[}:]` so `{codex}`/`{codeword}` no longer match while `{code}` and `{code:lang}` still do.

File: skills/jira-syntax/scripts/validate-jira-syntax.sh [hunk 4]
```diff
@@ -104,17 +104,21 @@ validate_file() {
...
+        error "Mismatched {code} tags: odd number ($code_count) of occurrences (expected pairs)"
     fi
 
     # Check for unclosed {panel} blocks
-    local panel_open=$(echo "$content" | grep -c "{panel")
-    local panel_close=$(echo "$content" | grep -c "{/panel}")
-    if [ "$panel_open" -ne "$panel_close" ]; then
-        error "Mismatched {panel} tags: $panel_open opening, $panel_close closing"
+    # Same rule applies: {panel} opens and closes the block.
+    local panel_count=$(echo "$content" | grep -o "{panel" | wc -l)
```
[the remark this answers is not in the document]
Comment (powli) [reply skills/jira-syntax/scripts/validate-jira-syntax.sh]: Fixed in 9bcfa49: switched to here-string for the {panel} count (also tightened regex to `\{panel[}:]`).

File: skills/jira-syntax/scripts/validate-jira-syntax.sh [hunk 5]
```diff
@@ -104,17 +104,21 @@ validate_file() {
...
-    if [ "$code_open" -ne "$code_close" ]; then
-        error "Mismatched {code} tags: $code_open opening, $code_close closing"
+    # Jira wiki markup uses {code} as both the opening and closing tag, so a
+    # correctly paired block always produces an even occurrence count.
+    # Use `grep -o ... | wc -l` to count each occurrence (not just matching
+    # lines), matching the {color} check below for consistency and to catch
+    # multiple tags on the same line.
+    local code_count=$(echo "$content" | grep -o "{code" | wc -l)
+    if [ $((code_count % 2)) -ne 0 ]; then
+        error "Mismatched {code} tags: odd number ($code_count) of occurrences (expected pairs)"
```
[the remark this answers is not in the document]
Comment (powli) [reply skills/jira-syntax/scripts/validate-jira-syntax.sh]: Fixed in 9bcfa49: switched to here-string for the {code} count (also tightened regex to `\{code[}:]`).

File: skills/jira-syntax/scripts/validate-jira-syntax.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (powli) [reply skills/jira-syntax/scripts/validate-jira-syntax.sh]: Fixed in 9bcfa49: same treatment for the {panel} check — here-string plus `\{panel[}:]` regex tighten.
